### PR TITLE
[ADD] stock_picking_batch_validate_confirm

### PR DESCRIPTION
**Pull request by EmilioPascual**
Wizard to confirm batch if pending origin moves exists when validate batch.

@yajo @Shide @rafaelbn @Gelojr please review.

MT-5994 @moduon

### CODE REVIEW

File: stock_picking_batch_validate_confirm/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+{
+    "name": "Stock Picking Batch Validate Confirm",
+    "summary": "Request confirmation when validate batch if there are pending origin moves",
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/__manifest__.py]:
```suggestion
    "summary": "Request confirmation when validating batch if there are pending origin moves",
```

File: stock_picking_batch_validate_confirm/readme/CONTEXT.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+This module was developed because 
+
+It will be useful for you if ...
+
+If you need this module for those reasons, these might interest you too:
+
+-   ...
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/readme/CONTEXT.md]: issue(docs): add context or remove file please.

File: stock_picking_batch_validate_confirm/readme/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+1. Go to Inventory App > Configuration > Warehouses.
+1. Select warehouse or create new.
+1. Select *Send goods in output and then deliver (2 steps)*.
+1. Create some sale orders with storable product.
+1. Validate some picks generated by sale orders but not all.
+1. Go to Inventory App > Operations > Batch Transfer.
+1. Create new Batch Transfer and add deliveries (out) generated by sale orders.
+1. Set quantities done.
+1. *Validate* batch.
+1. A Wizard show pending origin moves, i.e. moves from picks not done yet.
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/readme/USAGE.md]:
```suggestion
1. A Wizard shows pending origin moves, i.e. moves from picks not done yet.
```

File: stock_picking_batch_validate_confirm/readme/USAGE.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+1. Click on *Validate* in wizard if you still want validate the batch.
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/readme/USAGE.md]:
```suggestion
1. Click on *Validate* in wizard if you still want to validate the batch.
```

File: stock_picking_batch_validate_confirm/readme/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+1. Click on *Cancel* in wizard if you don't want validate the batch untill all origin moves is done.
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/readme/USAGE.md]:
```suggestion
1. Click on *Cancel* in wizard if you don't want to validate the batch until all origin moves are done.
```

File: stock_picking_batch_validate_confirm/wizards/batch_confirm.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import fields, models
+
+
+class BatchConfirm(models.TransientModel):
+    _name = "batch.confirm"
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/wizards/batch_confirm.py]: suggestion: "batch" is a pretty wide term. I'd use a more pedantic name for the wizard:

```suggestion
class StockPickingBatchConfirm(models.TransientModel):
    _name = "stock.picking.batch.confirm"
```

File: stock_picking_batch_validate_confirm/wizards/batch_confirm.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+<!-- Copyright 2024 Moduon Team S.L.
+     License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0) -->
+<odoo>
+    <record id="batch_confirm_view_form" model="ir.ui.view">
+        <field name="name">batch.confirm.view.form</field>
+        <field name="model">batch.confirm</field>
+        <field name="arch" type="xml">
+            <form string="">
+                <p>You have <i>pending origin moves</i> yet, by clickcing on <i
+                    >validate</i> batches will be validated</p>
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/wizards/batch_confirm.xml]: suggestion: use a more friendly wording

```suggestion
                <p>There are still some <i>pending origin moves</i>. By clicking on <i
                    >validate</i>, batches will be validated</p>
```

File: stock_picking_batch_validate_confirm/wizards/batch_confirm.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+                <field name="batch_id" invisible="1" />
+                <field
+                    name="move_ids"
+                    readonly="1"
+                    context="{'tree_view_ref': 'stock_picking_batch.view_picking_move_tree_inherited'}"
+                    attrs="{'invisible': [('move_ids', '=', False)]}"
```
Comment (yajo) [inline stock_picking_batch_validate_confirm/wizards/batch_confirm.xml]: thought: this case should never happen, right?

```suggestion
```

**Review by yajo — CHANGES_REQUESTED** [review]
Code review.

**Comment by EmilioPascual** [comment]
Thanks @yajo for your review, all request are resolved.

**Review by yajo — APPROVED** [review]
Excellent, thanks!

**Comment by EmilioPascual** [comment]
I have recorded a video with behavior:
https://www.loom.com/share/f12d566c131c43a2873d84d550306254?sid=7c54113d-abc2-4f3c-8bd5-e38bf2df234d

**Review by Shide — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
I've seen the video 

👍🏼

**Pull request merged**